lib/agent.js

```diff
@@ -775,46 +755,12 @@ function _addIntrinsicAttrsFromTransaction(transaction) {
       transaction.getFullName(),
       transaction.referringPathHash
     )
-    if (transaction.referringPathHash) {
-      intrinsicAttributes['nr.referringPathHash'] = transaction.referringPathHash
-    }
-    if (transaction.referringTransactionGuid) {
-      const refId = transaction.referringTransactionGuid
-      intrinsicAttributes['nr.referringTransactionGuid'] = refId
-    }
-    const alternatePathHashes = transaction.alternatePathHashes()
-    if (alternatePathHashes) {
-      intrinsicAttributes['nr.alternatePathHashes'] = alternatePathHashes
-    }
-    if (transaction.baseSegment && transaction.type === 'web') {
-      const apdex =
-        this.config.web_transactions_apdex[transaction.getFullName()] || this.config.apdex_t
-      const duration = transaction.baseSegment.getDurationInMillis() / 1000
-      intrinsicAttributes['nr.apdexPerfZone'] = calculateApdexZone(duration, apdex)
-    }
-  }
-
-  if (transaction.syntheticsData) {
-    intrinsicAttributes['nr.syntheticsResourceId'] = transaction.syntheticsData.resourceId
-    intrinsicAttributes['nr.syntheticsJobId'] = transaction.syntheticsData.jobId
-    intrinsicAttributes['nr.syntheticsMonitorId'] = transaction.syntheticsData.monitorId
+    AttributeHelper.maybeAddCATAttributes(transaction, intrinsicAttributes, this.config)
```

Comment:
it feels odd that's there one sequence left of setting attrs inline whereas the rest were abstracted and moved to other functions. thoughts on moving this to its own function or moving this code into maybeAddCATAttributes?